### ISSUE TITLE
V-sync and interaction button problem

### ISSUE THREAD

**Issue by medion6** [SuiMachine/Deadly-Premonition-2---Sui-s-hack#12]
I have a strange problem too that with vsync activated sometimes it skips the interaction button presses (b by default, but i tried swapping it too) and does not record them properly. Which results in you sometimes having to tap a few times to interact with objects on the map or other characters, seems random. It only happens when you control the character and it doesn't seem to happen in the menus. I have solved it by disabling vsync. I dont use any other unity 50fps camera fixes or mods. Im using the official Xbox Elite 2 V2 controller and my monitor is a Dell 2k 165hz.

Edit: I still have the problem with vsync off from time to time, I don't know what it can be. I dont think its a problem with the hack itself, it seems fps cap or hz related, the red room menu cursor was speed up too. Only game where all this happens although I experienced something similar unlocking Nier Automata fps with Special K because some actions are tied to the framerate. Only fix that seems to work right now is fps cap at 60fps, could you make it possible to enter the fps cap values ​​manually like the monitor hz? That would help a lot because its hard to select what you need now with the mouse slider only. Thanks.  Any other ideas around it? I cant go over 60fps (tried 120 too) right now without getting this issue.

**Comment by SuiMachine** [comment]
Unless you go exclusive fullscreen - Unity uses monitor refresh rate (if Vsync is enabled) and ignores target framerate. If you disable V-sync, then target framerate is used and you can configure it to whatever you want - technically if you use borderless it will likely result in somewhat odd frame pacing, but shouldn't introduce tearing, as this game is not using DXGI Flip Model.

As for the input being ignored - again, not something I have experienced, with my game being capped at 144. Unity itself processes inputs per frame, unless for whatever reason, devs write some inputs to be handled in FixedUpdates. But with SteamInput added on top, maybe some mistakes were made.

**Comment by medion6** [comment]
Thanks for the input. I can't seem to get around this problem either, maybe because my monitor is the odd dog at 165hz? In the end the only solution for me is to play at caped 60fps.

**Comment by medion6** [comment]
Yes, sometimes you have to move against the object for it to detect them or even wait for a character to look back at you even at 60fps vanilla game. But this exact problem is easily recognizable from that because sometimes you have to hit the action button up to 3 or 5 times to enter a vending machine (for example) in front of it and other occasions it works just fine.
